lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_node_lowering.cpp

```diff
@@ -58,8 +60,33 @@ class TSNodeLowering : public NodeLowering {
       return false;
     }
     LTC_CHECK_EQ(node->num_outputs(), ops.size());
-    for (size_t i = 0; i < ops.size(); ++i) {
-      loctx()->AssignOutputOp(ir::Output(node, i), ops[i]);
+    if (lazy_tensors::sys_util::GetEnvBool("LTC_IR_TS_SHAPE_TYPES", false)) {
+      for (size_t i = 0; i < ops.size(); ++i) {
+        loctx()->AssignOutputOp(ir::Output(node, i), ops[i]);
+      }
+    } else {
+      auto type_transformer = [](const Shape& shape, const at::TypePtr& type) {
+        return type->expect<at::TensorType>()
+            ->withScalarType(TensorTypeFromLtcType(shape.element_type()))
+            ->withSizesStrides(shape.as_sizes(), ComputeShapeStrides(shape));
```

Comment:
not sure if this covers everything (though it does like it does from the TS IR generated)

Reply:
hhm, I'm not seeing here
* `device`
* `requires_grad` (which we may not care about)

You could use one of `TensorType::create` constructors to avoid constructing a type, one property at a time.

` ComputeShapeStrides(shape)` do we know if this handles transposed strides correctly?

Reply:
my thought in using the `with*` properties was to preserve everything about the original type except for what I can get from the shape. so e.g. `device` and `requires_grad` should be preserved right? 

https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/aten/src/ATen/core/jit_type.h#L569

The fact that those attributes aren't rendered in the type is due to how the JIT serializes the graph I think? 

https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/backend_impl.cpp#L70

re `ComputeShapeStrides` i'm not sure but maybe `minor_to_major` here implies yes?

https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/lazy_tensor_core/lazy_tensor_core/csrc/tensor_util.cpp#L704

Reply:
FWIW I don't think we'd have a use for either in the generated TS graphs, we use TS-as-a-backend here. The model of operation is that the device stays the same as the inputs device and we don't care about gradients either at this point, it's just a computation backend at this point and differentiation has been taken care of by autograd, TL;DR I wouldn't worry about preserving anything but shapes.